src/modules/html/producer/html_producer.cpp

```diff
@@ -361,7 +371,11 @@ class html_client
         auto name = message->GetName().ToString();
 
         if (name == REMOVE_MESSAGE_NAME) {
-            // TODO
+            // TODO fully remove producer
+            this->close();
+            last_frame_ = core::draw_frame::empty();
```

Comment:
Thinking about this use of `last_frame_` I dont think it is reliable enough. If there are any frames left in `frames_` then your override here will quickly be lost.

Either `frames_` should also be emptied (and risk losing the last frame or two of an animation).
or this could be changed to allow any buffered frames to play before replacing with empty
```
{
    std::lock_guard<std::mutex> lock(frames_mutex_);
    frames_.push(draw_frame::empty());
}
```


Also you are missing a use of `last_frame_mutex_` here.

Reply:
Sounds good.